01-import_and_filter.py

```diff
@@ -18,20 +18,58 @@
 
 import mne
 from mne.parallel import parallel_func
+from warnings import warn
+import sys
 
+# import glob
 import config
 
 
 def run_filter(subject):
     print("processing subject: %s" % subject)
-    # XXX : put the study-specific names in the config file
+
     meg_subject_dir = op.join(config.meg_dir, subject)
-    raw_fnames_in = [op.join(meg_subject_dir, '%s_audvis_raw.fif' % subject)]
-    raw_fnames_out = [
-        op.join(meg_subject_dir, '%s_audvis_filt_raw.fif' % subject)]
 
+    raw_fnames_in = list()
+    raw_fnames_out = list()
+
+    # input files
+    if config.runs:
+        base_raw_fname_in = '{subject}_{study_name}_{run}_raw.fif'
+        base_raw_fname_out = '{subject}_{study_name}_{run}_filt_raw.fif'
+
+        for run in config.runs:
+            raw_fnames_in.append(base_raw_fname_in.format(run=run,
+                                                          study_name=config.study_name,
+                                                          subject=subject))
+
+            raw_fnames_out.append(base_raw_fname_out.format(run=run,
+                                                            study_name=config.study_name,
+                                                            subject=subject))
+
+    else:
+        base_raw_fname_in = '{subject}_{study_name}_raw.fif'
+        base_raw_fname_out = '{subject}_{study_name}_filt_raw.fif'
+
+        raw_fnames_in.append(base_raw_fname_in.format(
+            study_name=config.study_name, subject=subject))
+        raw_fnames_out.append(base_raw_fname_out.format(
+            study_name=config.study_name, subject=subject))
+
+    raws = []
+    print("Try loading %d files" % len(raw_fnames_in))
     for raw_fname_in, raw_fname_out in zip(raw_fnames_in, raw_fnames_out):
-        raw = mne.io.read_raw_fif(raw_fname_in, preload=True, verbose='error')
+        print(raw_fname_in)
+        try:
+            raw = mne.io.read_raw_fif(op.join(meg_subject_dir, raw_fname_in),
+                                      preload=True, verbose='error')
+        except FileNotFoundError:
+            if not raws:
+                raise ValueError('Cannot find ' + raw_fname_in)
+            if raws:
+                runs_found = len(raws)
+                warn('Found only ' + str(runs_found) + ' runs')
+            break
```

Comment:
I would raise a warning for each raw not found. Nothing fancy.

```
raw_fname_path = op.join(meg_subject_dir, raw_fname_in)
if not op.exists(raw_fname_path):
     warn('Run %s not found for subject %s ' % (raw_fname_in, subject))
raw = mne.io.read_raw_fif(raw_fname_path, preload=True, verbose='error')
```

this avoids a try which is better

Reply:
code above updated.

Reply:
Ok, done. Still, I kept the error when no file can be found. 
`raise ValueError('Cannot find ' + raw_fname_path)`

The warning message still looks murky on the command line. I don't get why it prints the command again:
> /Users/sophie/repos/mne-study-template/01-import_and_filter.py:68: UserWarning: Run sample_audvis_run02_raw.fif not found for subject sample 
>   warn('Run %s not found for subject %s ' % (raw_fname_in, subject))